packages/flutter_test/lib/src/finders.dart

```diff
@@ -365,13 +365,18 @@ class CommonFinders {
   ///
   /// ```dart
   /// expect(find.byTooltip('Back'), findsOneWidget);
+  /// expect(find.byTooltip(RegExp('Back.*')), findsNWidgets(2));
   /// ```
   ///
   /// If the `skipOffstage` argument is true (the default), then this skips
   /// nodes that are [Offstage] or that are from inactive [Route]s.
-  Finder byTooltip(String message, { bool skipOffstage = true }) {
+  Finder byTooltip(Pattern message, { bool skipOffstage = true }) {
     return byWidgetPredicate(
-      (Widget widget) => widget is Tooltip && widget.message == message,
+      (Widget widget) {
+        return widget is Tooltip && (message is RegExp
+            ? message.hasMatch(widget.message ?? '')
+            : message == widget.message);
+      },
```

Comment:
Unrelated to this PR, just an observation: Looks like this matcher is completely ignoring tooltips that have been configured with `Tooltip.richMessage`, https://main-api.flutter.dev/flutter/material/Tooltip/richMessage.html.

Reply:
Ooh, good catch.  I'll add that.